web/packages/teleport/src/MFAContext/MFAContext.tsx

```diff
@@ -0,0 +1,45 @@
+import { PropsWithChildren, createContext, useCallback, useRef } from 'react';
+import AuthnDialog from 'teleport/components/AuthnDialog';
+import { useMfa } from 'teleport/lib/useMfa';
+import auth, { MfaChallengeScope } from 'teleport/services/auth/auth';
+import { MfaChallengeResponse } from 'teleport/services/mfa';
+
+export interface MfaContextValue {
+  getAdminActionMfaResponse(reusable?: boolean): Promise<MfaChallengeResponse>;
+}
+
+export const MfaContext = createContext<MfaContextValue>(null);
+
+export const MfaContextProvider = ({ children }: PropsWithChildren) => {
+  const allowReuse = useRef(false);
+  const adminMfa = useMfa({
+    req: {
+      scope: MfaChallengeScope.ADMIN_ACTION,
+      allowReuse: allowReuse.current,
+      isMfaRequiredRequest: {
+        admin_action: {},
+      },
+    },
+  });
+
+  const getAdminActionMfaResponse = useCallback(
+    async (reusable: boolean = false) => {
+      allowReuse.current = reusable;
+      return (await adminMfa.getChallengeResponse()) || {}; // return an empty challenge to prevent mfa retry.
```

Comment:
```suggestion
      const res = await adminMfa.getChallengeResponse();
      
      if (!res) {
        return {}; // return an empty challenge to prevent mfa retry.
      }
      
      return res;
```